src/components/Vaccinations.tsx

```diff
@@ -1,13 +1,16 @@
-'use client'
+'use client';
+
 import React, { useState } from 'react';
 import { supabase } from '@/services/supabase';
+import { Loader } from '@/components/Loader';
 
 interface VaccinationsProps {
   pet_id: string;
 }
 
 export const Vaccinations: React.FC<VaccinationsProps> = ({ pet_id }) => {
   const [vaccinations, setVaccinations] = useState<any[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
 
```

Comment:
🟡 I suggest implementing a similar approach to what I mentioned for the Medications component. Specifically, consider adding `const [errorMessage, setErrorMessage] = useState<string | null>(null);` to handle and display error messages to the user. This way, instead of just showing an infinite loading state, users will receive a clear message if an error occurs during data fetching.

Reply:
Great advice!

I removed unnecessary Suspense from PetTabs (however I am still unsure what the best practices are), fixed the infinite loop in Medications - thanks fo much for catching that! i didn't notice it at all - and added browser alerts for failed fetches for Vaccinations and Medications.

Maybe we can design an error component to display to the user when database call error occurs? Like some (funny) image (svg?) and error text?

Reply:
Yes, a separate error component with nice design sounds great! I will work on that in Figma